internal/runners/activate/activate.go

```diff
@@ -49,7 +53,28 @@ func (r *Activate) Run(params *ActivateParams) error {
 }
 
 func (r *Activate) run(params *ActivateParams, activatorLoop activationLoopFunc) error {
-	logging.Debug("Activate %v, %v", params.Namespace, params.PreferredPath)
+	logging.Debug("Activate with namespace=%v, path=%v", params.Namespace, params.PreferredPath)
+
+	activeProjectDir := os.Getenv(constants.ActivatedStateEnvVarName)
+	alreadyActivated := activeProjectDir != ""
+
+	// handle case, if we are already activated
+	if alreadyActivated {
+		if params.Default && params.Namespace.IsValid() {
+			return locale.NewError("err_default_with_name_already_active", "Trying to set {{.V0}} as default project while in activated environment.  Please de-activate the current runtime.", params.Namespace.String())
```

Comment:
We should still set the default. This is a legitimate use-case we want to facilitate.

The AC's specifically state this:

> When called in an already activated environment, only set up the default (don't activate again or error out about already being activated)

Reply:
Yes, this only for the case where a name has been explicitly specified ie.,

You are in an activated state for project `testOrg/A` and then you run `state activate testOrg/B --default`.  As it is specifically requested, probably `testOrg/B` should have been selected, but after a few initial trials it became to confusing too me how to ensure that going forward we don't use `testOrg/A`.  As this, I think, is also a target of your refactoring, it will hopefully be easier after the refactoring.

If you just run `state activate --default` in an activated project, it would work as requested in the AC.